src/main/java/com/example/teamboolean/apprentidash/ApprentiDashController.java

```diff
@@ -116,49 +114,60 @@ public String recordHour(Model m, Principal p){
         loggedInStatusHelper(m, p);
         m.addAttribute("currentPage", "clock_in");
         //Sets status for knowing which button to show
-        String todayDate =  java.time.format.DateTimeFormatter.RFC_1123_DATE_TIME.format(ZonedDateTime.now(ZoneOffset.UTC));
-        m.addAttribute("workStatus", buttonRenderHelper());
-        m.addAttribute("todayDate", todayDate);
+        LocalDateTime now = LocalDateTime.now();
+        AppUser currentUser = userRepository.findByUsername(p.getName());
+        m.addAttribute("workStatus", buttonRenderHelper(currentUser));
+        m.addAttribute("todayDate", now);
         return "recordHour";
     }
 
 //Route to handle our clock in button
     @PostMapping(value="/recordHour", params="name=value")
     public String clockInSave(Principal p, Model m) {
-        DateTimeFormatter dtf = DateTimeFormatter.ofPattern("yyyy/MM/dd HH:mm:ss");
-        LocalDateTime now = LocalDateTime.now();
-        boolean endOfDay = false;
-
-        if(buttonRenderHelper().equals("clockIn")) {
-            currentDay.setClockIn(now);
-        }else if(buttonRenderHelper().equals(("lunchIn"))) {
-            currentDay.setLunchStart(now);
-        }else if(buttonRenderHelper().equals("lunchOut")) {
-            currentDay.setLunchEnd(now);
-        }else if(buttonRenderHelper().equals("clockOut")){
-            currentDay.setClockOut(now);
-            endOfDay = true;
-        }
-        currentDay.setUser(userRepository.findByUsername(p.getName()));
-        dayRepository.save(currentDay);
-        if(endOfDay){
-            currentDay = new Day();
+
+        AppUser currentUser = userRepository.findByUsername(p.getName());
+
+        if(buttonRenderHelper(currentUser).equals("clockIn")) {
+            currentUser.getCurrentday().setClockIn(LocalDateTime.now());
+        }else if(buttonRenderHelper(currentUser).equals(("lunchIn"))) {
+            currentUser.getCurrentday().setLunchStart(LocalDateTime.now());
+        }else if(buttonRenderHelper(currentUser).equals("lunchOut")) {
+            currentUser.getCurrentday().setLunchEnd(LocalDateTime.now());
+        }else if(buttonRenderHelper(currentUser).equals("clockOut")){
+            currentUser.getCurrentday().setClockOut(LocalDateTime.now());
         }
-        m.addAttribute("workStatus", buttonRenderHelper());
+
+        currentUser.getCurrentday().setUser(currentUser);
```

Comment:
What is this line doing? 
If we are getting a currentDay from the user object then why are we setting the day's user to current user? Wouldn't it already be associated with the user?

Reply:
Done